In [163]:
import pandas as pd
df_original = pd.read_excel("datasets/kaggle_data_moneyball_baseball_1999_2012.xlsx",header=1)
df = df_original
display(df)


Team League  Year   RS   RA   W    OBP    SLG     BA  Playoffs  \
0    ARI     NL  2012  734  688  81  0.328  0.418  0.259         0   
1    ATL     NL  2012  700  600  94  0.320  0.389  0.247         1   
2    BAL     AL  2012  712  705  93  0.311  0.417  0.247         1   
3    BOS     AL  2012  734  806  69  0.315  0.415  0.260         0   
4    CHC     NL  2012  613  759  61  0.302  0.378  0.240         0   
..   ...    ...   ...  ...  ...  ..    ...    ...    ...       ...   
415  SFG     NL  1999  872  831  86  0.356  0.434  0.271         0   
416  STL     NL  1999  809  838  75  0.338  0.426  0.262         0   
417  TBD     AL  1999  772  913  69  0.343  0.411  0.274         0   
418  TEX     AL  1999  945  859  95  0.361  0.479  0.293         1   
419  TOR     AL  1999  883  862  84  0.352  0.457  0.280         0   

     RankSeason  RankPlayoffs    G   OOBP   OSLG  
0           NaN           NaN  162  0.317  0.415  
1           4.0           5.0  162  0.306  0.378  
2           5.0           4.0  162  0.315  0.403  
3           NaN           NaN  162  0.331  0.428  
4           NaN           NaN  162  0.335  0.424  
..          ...           ...  ...    ...    ...  
415         NaN           NaN  162  0.345  0.423  
416         NaN           NaN  161  0.355  0.427  
417         NaN           NaN  162  0.371  0.448  
418         5.0           4.0  162  0.346  0.459  
419         NaN           NaN  162  0.353  0.456  

[420 rows x 15 columns]

In [164]:
df.corr(numeric_only=True)

Year        RS        RA         W       OBP       SLG  \
Year          1.000000 -0.394162 -0.375678  0.002692 -0.402147 -0.349168   
RS           -0.394162  1.000000  0.198869  0.554444  0.895743  0.910242   
RA           -0.375678  0.198869  1.000000 -0.634540  0.155346  0.213009   
W             0.002692  0.554444 -0.634540  1.000000  0.522941  0.497791   
OBP          -0.402147  0.895743  0.155346  0.522941  1.000000  0.773883   
SLG          -0.349168  0.910242  0.213009  0.497791  0.773883  1.000000   
BA           -0.329479  0.806596  0.232224  0.393387  0.829507  0.754388   
Playoffs      0.017266  0.390044 -0.433023  0.709942  0.387661  0.335586   
RankSeason    0.080417 -0.360962  0.069920 -0.764191 -0.264424 -0.268387   
RankPlayoffs  0.047934 -0.001477  0.053172 -0.004507 -0.047524 -0.077998   
G             0.102071  0.127402 -0.037377  0.129136  0.083441  0.110088   
OOBP         -0.371869  0.133981  0.910344 -0.634315  0.120349  0.143861   
OSLG         -0.345667  0.193426  0.909319 -0.568490  0.153372  0.210423   

                    BA  Playoffs  RankSeason  RankPlayoffs         G  \
Year         -0.329479  0.017266    0.080417      0.047934  0.102071   
RS            0.806596  0.390044   -0.360962     -0.001477  0.127402   
RA            0.232224 -0.433023    0.069920      0.053172 -0.037377   
W             0.393387  0.709942   -0.764191     -0.004507  0.129136   
OBP           0.829507  0.387661   -0.264424     -0.047524  0.083441   
SLG           0.754388  0.335586   -0.268387     -0.077998  0.110088   
BA            1.000000  0.285183   -0.164299     -0.108597  0.099764   
Playoffs      0.285183  1.000000         NaN           NaN  0.054194   
RankSeason   -0.164299       NaN    1.000000     -0.007525  0.065533   
RankPlayoffs -0.108597       NaN   -0.007525      1.000000 -0.011897   
G             0.099764  0.054194    0.065533     -0.011897  1.000000   
OOBP          0.187108 -0.433572    0.097731      0.065808 -0.077055   
OSLG          0.227808 -0.366744    0.050111      0.087089 -0.004188   

                  OOBP      OSLG  
Year         -0.371869 -0.345667  
RS            0.133981  0.193426  
RA            0.910344  0.909319  
W            -0.634315 -0.568490  
OBP           0.120349  0.153372  
SLG           0.143861  0.210423  
BA            0.187108  0.227808  
Playoffs     -0.433572 -0.366744  
RankSeason    0.097731  0.050111  
RankPlayoffs  0.065808  0.087089  
G            -0.077055 -0.004188  
OOBP          1.000000  0.830902  
OSLG          0.830902  1.000000

In the interest of getting a good price, we are going to exclude any teams who have been in the Playoffs

In [165]:
df.pop('G') #All the same
df.drop(df[df['Playoffs'] == 1].index,inplace=True)
df.drop(df[df['RankPlayoffs'] == 1].index,inplace=True)
display(df)


Team League  Year   RS   RA   W    OBP    SLG     BA  Playoffs  \
0    ARI     NL  2012  734  688  81  0.328  0.418  0.259         0   
3    BOS     AL  2012  734  806  69  0.315  0.415  0.260         0   
4    CHC     NL  2012  613  759  61  0.302  0.378  0.240         0   
5    CHW     AL  2012  748  676  85  0.318  0.422  0.255         0   
7    CLE     AL  2012  667  845  68  0.324  0.381  0.251         0   
..   ...    ...   ...  ...  ...  ..    ...    ...    ...       ...   
414  SEA     AL  1999  859  905  79  0.343  0.455  0.269         0   
415  SFG     NL  1999  872  831  86  0.356  0.434  0.271         0   
416  STL     NL  1999  809  838  75  0.338  0.426  0.262         0   
417  TBD     AL  1999  772  913  69  0.343  0.411  0.274         0   
419  TOR     AL  1999  883  862  84  0.352  0.457  0.280         0   

     RankSeason  RankPlayoffs   OOBP   OSLG  
0           NaN           NaN  0.317  0.415  
3           NaN           NaN  0.331  0.428  
4           NaN           NaN  0.335  0.424  
5           NaN           NaN  0.319  0.405  
7           NaN           NaN  0.336  0.430  
..          ...           ...    ...    ...  
414         NaN           NaN  0.368  0.454  
415         NaN           NaN  0.345  0.423  
416         NaN           NaN  0.355  0.427  
417         NaN           NaN  0.371  0.448  
419         NaN           NaN  0.353  0.456  

[306 rows x 14 columns]

RA,OOBP, and OSLG have the strongest negative correlation for wins, so we are going to purge the teams with valeus lower than the 3rd quartile

In [166]:
ra_mean = df['RA'].quantile(.75)
oobp_mean = df['OOBP'].quantile(.75)
oslg_mean = df['OSLG'].quantile(.75)
print(f'RA 3rd Quartile: {ra_mean}\n OOBP 3rd Quartile: {oobp_mean}\n OSlG 3rd Quartile: {oslg_mean}')

RA 3rd Quartile: 844.0
 OOBP 3rd Quartile: 0.346
 OSlG 3rd Quartile: 0.443


In [167]:
df.drop(df[df['RA'] <= ra_mean].index,inplace=True)
df.drop(df[df['OOBP'] <= oobp_mean].index,inplace=True)
df.drop(df[df['OSLG'] <= oslg_mean].index,inplace=True)
print(df.shape)
display(df)

(52, 14)


Team League  Year   RS    RA   W    OBP    SLG     BA  Playoffs  \
8    COL     NL  2012  758   890  64  0.330  0.436  0.274         0   
81   PIT     NL  2010  587   866  57  0.304  0.373  0.242         0   
92   BAL     AL  2009  741   876  64  0.332  0.415  0.268         0   
119  WSN     NL  2009  710   874  59  0.337  0.406  0.258         0   
141  PIT     NL  2008  735   884  67  0.320  0.403  0.258         0   
147  TEX     AL  2008  901   967  79  0.354  0.462  0.283         0   
171  PIT     NL  2007  724   846  68  0.325  0.411  0.263         0   
176  TBD     AL  2007  782   944  66  0.336  0.433  0.268         0   
182  BAL     AL  2006  768   899  70  0.339  0.424  0.277         0   
192  KCR     AL  2006  757   971  62  0.332  0.411  0.271         0   
206  TBD     AL  2006  689   856  61  0.314  0.420  0.255         0   
209  WSN     NL  2006  746   872  71  0.338  0.418  0.262         0   
216  CIN     NL  2005  820   889  73  0.339  0.446  0.261         0   
218  COL     NL  2005  740   862  67  0.333  0.411  0.267         0   
222  KCR     AL  2005  701   935  56  0.320  0.396  0.263         0   
236  TBD     AL  2005  750   936  67  0.329  0.425  0.274         0   
247  CIN     NL  2004  750   907  76  0.331  0.418  0.250         0   
249  COL     NL  2004  833   923  68  0.345  0.455  0.275         0   
253  KCR     AL  2004  720   905  58  0.322  0.397  0.259         0   
277  CIN     NL  2003  694   886  69  0.318  0.395  0.245         0   
279  COL     NL  2003  853   892  74  0.344  0.445  0.267         0   
280  DET     AL  2003  591   928  43  0.300  0.375  0.240         0   
283  KCR     AL  2003  836   867  83  0.336  0.427  0.274         0   
285  MIL     NL  2003  714   873  68  0.329  0.419  0.256         0   
298  TEX     AL  2003  826   969  71  0.330  0.454  0.266         0   
309  COL     NL  2002  778   898  73  0.337  0.423  0.274         0   
313  KCR     AL  2002  737   891  62  0.323  0.398  0.256         0   
327  TBD     AL  2002  673   918  55  0.314  0.390  0.253         0   
328  TEX     AL  2002  843   882  72  0.338  0.455  0.269         0   
339  COL     NL  2001  923   906  73  0.354  0.483  0.292         0   
340  DET     AL  2001  724   876  66  0.320  0.409  0.260         0   
343  KCR     AL  2001  729   858  65  0.318  0.409  0.266         0   
358  TEX     AL  2001  890   968  73  0.344  0.471  0.275         0   
360  ANA     AL  2000  864   869  82  0.352  0.472  0.280         0   
363  BAL     AL  2000  794   913  74  0.341  0.435  0.272         0   
365  CHC     NL  2000  764   904  65  0.335  0.411  0.256         0   
369  COL     NL  2000  968   897  82  0.362  0.455  0.294         0   
372  HOU     NL  2000  938   944  72  0.361  0.477  0.278         0   
373  KCR     AL  2000  879   930  77  0.348  0.425  0.288         0   
376  MIN     AL  2000  748   880  69  0.337  0.407  0.270         0   
377  MON     NL  2000  738   902  67  0.326  0.432  0.266         0   
388  TEX     AL  2000  848   974  71  0.352  0.446  0.283         0   
389  TOR     AL  2000  861   908  83  0.341  0.469  0.275         0   
396  CHW     AL  1999  777   870  75  0.337  0.429  0.277         0   
399  COL     NL  1999  906  1028  72  0.348  0.472  0.288         0   
400  DET     AL  1999  747   882  69  0.326  0.443  0.261         0   
403  KCR     AL  1999  856   921  64  0.348  0.433  0.282         0   
405  MIL     NL  1999  815   886  74  0.353  0.426  0.273         0   
411  PHI     NL  1999  841   846  77  0.351  0.431  0.275         0   
414  SEA     AL  1999  859   905  79  0.343  0.455  0.269         0   
417  TBD     AL  1999  772   913  69  0.343  0.411  0.274         0   
419  TOR     AL  1999  883   862  84  0.352  0.457  0.280         0   

     RankSeason  RankPlayoffs   OOBP   OSLG  
8           NaN           NaN  0.357  0.470  
81          NaN           NaN  0.348  0.449  
92          NaN           NaN  0.353  0.476  
119         NaN           NaN  0.352  0.450  
141       

In [168]:
df.corr(numeric_only=True)

Year        RS        RA         W       OBP       SLG  \
Year          1.000000 -0.456660 -0.112363 -0.411673 -0.418131 -0.363089   
RS           -0.456660  1.000000  0.341903  0.780816  0.915712  0.890766   
RA           -0.112363  0.341903  1.000000 -0.050909  0.239561  0.332611   
W            -0.411673  0.780816 -0.050909  1.000000  0.746948  0.712645   
OBP          -0.418131  0.915712  0.239561  0.746948  1.000000  0.780452   
SLG          -0.363089  0.890766  0.332611  0.712645  0.780452  1.000000   
BA           -0.369095  0.855426  0.320244  0.636537  0.846945  0.734731   
Playoffs           NaN       NaN       NaN       NaN       NaN       NaN   
RankSeason         NaN       NaN       NaN       NaN       NaN       NaN   
RankPlayoffs       NaN       NaN       NaN       NaN       NaN       NaN   
OOBP         -0.065184  0.310973  0.663687 -0.000428  0.254442  0.278829   
OSLG          0.007125  0.344493  0.621121  0.015437  0.225905  0.323735   

                    BA  Playoffs  RankSeason  RankPlayoffs      OOBP      OSLG  
Year         -0.369095       NaN         NaN           NaN -0.065184  0.007125  
RS            0.855426       NaN         NaN           NaN  0.310973  0.344493  
RA            0.320244       NaN         NaN           NaN  0.663687  0.621121  
W             0.636537       NaN         NaN           NaN -0.000428  0.015437  
OBP           0.846945       NaN         NaN           NaN  0.254442  0.225905  
SLG           0.734731       NaN         NaN           NaN  0.278829  0.323735  
BA            1.000000       NaN         NaN           NaN  0.385523  0.236651  
Playoffs           NaN       NaN         NaN           NaN       NaN       NaN  
RankSeason         NaN       NaN         NaN           NaN       NaN       NaN  
RankPlayoffs       NaN       NaN         NaN           NaN       NaN       NaN  
OOBP          0.385523       NaN         NaN           NaN  1.000000  0.361493  
OSLG          0.236651       NaN         NaN           NaN  0.361493  1.000000

Wins have a high correlation with OBP for the remaining teams, so we are going to prioritize teams with high values in this area

In [169]:
df.sort_values('OBP',ascending=False,inplace=True)
display(df)

Team League  Year   RS    RA   W    OBP    SLG     BA  Playoffs  \
369  COL     NL  2000  968   897  82  0.362  0.455  0.294         0   
372  HOU     NL  2000  938   944  72  0.361  0.477  0.278         0   
339  COL     NL  2001  923   906  73  0.354  0.483  0.292         0   
147  TEX     AL  2008  901   967  79  0.354  0.462  0.283         0   
405  MIL     NL  1999  815   886  74  0.353  0.426  0.273         0   
419  TOR     AL  1999  883   862  84  0.352  0.457  0.280         0   
388  TEX     AL  2000  848   974  71  0.352  0.446  0.283         0   
360  ANA     AL  2000  864   869  82  0.352  0.472  0.280         0   
411  PHI     NL  1999  841   846  77  0.351  0.431  0.275         0   
399  COL     NL  1999  906  1028  72  0.348  0.472  0.288         0   
373  KCR     AL  2000  879   930  77  0.348  0.425  0.288         0   
403  KCR     AL  1999  856   921  64  0.348  0.433  0.282         0   
249  COL     NL  2004  833   923  68  0.345  0.455  0.275         0   
358  TEX     AL  2001  890   968  73  0.344  0.471  0.275         0   
279  COL     NL  2003  853   892  74  0.344  0.445  0.267         0   
417  TBD     AL  1999  772   913  69  0.343  0.411  0.274         0   
414  SEA     AL  1999  859   905  79  0.343  0.455  0.269         0   
389  TOR     AL  2000  861   908  83  0.341  0.469  0.275         0   
363  BAL     AL  2000  794   913  74  0.341  0.435  0.272         0   
216  CIN     NL  2005  820   889  73  0.339  0.446  0.261         0   
182  BAL     AL  2006  768   899  70  0.339  0.424  0.277         0   
328  TEX     AL  2002  843   882  72  0.338  0.455  0.269         0   
209  WSN     NL  2006  746   872  71  0.338  0.418  0.262         0   
396  CHW     AL  1999  777   870  75  0.337  0.429  0.277         0   
376  MIN     AL  2000  748   880  69  0.337  0.407  0.270         0   
309  COL     NL  2002  778   898  73  0.337  0.423  0.274         0   
119  WSN     NL  2009  710   874  59  0.337  0.406  0.258         0   
283  KCR     AL  2003  836   867  83  0.336  0.427  0.274         0   
176  TBD     AL  2007  782   944  66  0.336  0.433  0.268         0   
365  CHC     NL  2000  764   904  65  0.335  0.411  0.256         0   
218  COL     NL  2005  740   862  67  0.333  0.411  0.267         0   
192  KCR     AL  2006  757   971  62  0.332  0.411  0.271         0   
92   BAL     AL  2009  741   876  64  0.332  0.415  0.268         0   
247  CIN     NL  2004  750   907  76  0.331  0.418  0.250         0   
8    COL     NL  2012  758   890  64  0.330  0.436  0.274         0   
298  TEX     AL  2003  826   969  71  0.330  0.454  0.266         0   
236  TBD     AL  2005  750   936  67  0.329  0.425  0.274         0   
285  MIL     NL  2003  714   873  68  0.329  0.419  0.256         0   
377  MON     NL  2000  738   902  67  0.326  0.432  0.266         0   
400  DET     AL  1999  747   882  69  0.326  0.443  0.261         0   
171  PIT     NL  2007  724   846  68  0.325  0.411  0.263         0   
313  KCR     AL  2002  737   891  62  0.323  0.398  0.256         0   
253  KCR     AL  2004  720   905  58  0.322  0.397  0.259         0   
141  PIT     NL  2008  735   884  67  0.320  0.403  0.258         0   
222  KCR     AL  2005  701   935  56  0.320  0.396  0.263         0   
340  DET     AL  2001  724   876  66  0.320  0.409  0.260         0   
343  KCR     AL  2001  729   858  65  0.318  0.409  0.266         0   
277  CIN     NL  2003  694   886  69  0.318  0.395  0.245         0   
206  TBD     AL  2006  689   856  61  0.314  0.420  0.255         0   
327  TBD     AL  2002  673   918  55  0.314  0.390  0.253         0   
81   PIT     NL  2010  587   866  57  0.304  0.373  0.242         0   
280  DET     AL  2003  591   928  43  0.300  0.375  0.240         0   

     RankSeason  RankPlayoffs   OOBP   OSLG  
369         NaN           NaN  0.354  0.471  
372         NaN           NaN  0.353  0.476  
339         NaN           NaN  0.350  0.480  
147         NaN           NaN  0.362  0.455  
405       

Teams with high winrates are going to (presumably) fetch higher prices, so we are now going to purge teams with exceptionally high wins

In [170]:
win_mean = df['W'].quantile(.75)
df.drop(df[df['W'] >= win_mean].index,inplace=True)
display(df)

Team League  Year   RS    RA   W    OBP    SLG     BA  Playoffs  \
372  HOU     NL  2000  938   944  72  0.361  0.477  0.278         0   
339  COL     NL  2001  923   906  73  0.354  0.483  0.292         0   
388  TEX     AL  2000  848   974  71  0.352  0.446  0.283         0   
399  COL     NL  1999  906  1028  72  0.348  0.472  0.288         0   
403  KCR     AL  1999  856   921  64  0.348  0.433  0.282         0   
249  COL     NL  2004  833   923  68  0.345  0.455  0.275         0   
358  TEX     AL  2001  890   968  73  0.344  0.471  0.275         0   
417  TBD     AL  1999  772   913  69  0.343  0.411  0.274         0   
216  CIN     NL  2005  820   889  73  0.339  0.446  0.261         0   
182  BAL     AL  2006  768   899  70  0.339  0.424  0.277         0   
328  TEX     AL  2002  843   882  72  0.338  0.455  0.269         0   
209  WSN     NL  2006  746   872  71  0.338  0.418  0.262         0   
376  MIN     AL  2000  748   880  69  0.337  0.407  0.270         0   
309  COL     NL  2002  778   898  73  0.337  0.423  0.274         0   
119  WSN     NL  2009  710   874  59  0.337  0.406  0.258         0   
176  TBD     AL  2007  782   944  66  0.336  0.433  0.268         0   
365  CHC     NL  2000  764   904  65  0.335  0.411  0.256         0   
218  COL     NL  2005  740   862  67  0.333  0.411  0.267         0   
192  KCR     AL  2006  757   971  62  0.332  0.411  0.271         0   
92   BAL     AL  2009  741   876  64  0.332  0.415  0.268         0   
8    COL     NL  2012  758   890  64  0.330  0.436  0.274         0   
298  TEX     AL  2003  826   969  71  0.330  0.454  0.266         0   
236  TBD     AL  2005  750   936  67  0.329  0.425  0.274         0   
285  MIL     NL  2003  714   873  68  0.329  0.419  0.256         0   
377  MON     NL  2000  738   902  67  0.326  0.432  0.266         0   
400  DET     AL  1999  747   882  69  0.326  0.443  0.261         0   
171  PIT     NL  2007  724   846  68  0.325  0.411  0.263         0   
313  KCR     AL  2002  737   891  62  0.323  0.398  0.256         0   
253  KCR     AL  2004  720   905  58  0.322  0.397  0.259         0   
141  PIT     NL  2008  735   884  67  0.320  0.403  0.258         0   
222  KCR     AL  2005  701   935  56  0.320  0.396  0.263         0   
340  DET     AL  2001  724   876  66  0.320  0.409  0.260         0   
343  KCR     AL  2001  729   858  65  0.318  0.409  0.266         0   
277  CIN     NL  2003  694   886  69  0.318  0.395  0.245         0   
206  TBD     AL  2006  689   856  61  0.314  0.420  0.255         0   
327  TBD     AL  2002  673   918  55  0.314  0.390  0.253         0   
81   PIT     NL  2010  587   866  57  0.304  0.373  0.242         0   
280  DET     AL  2003  591   928  43  0.300  0.375  0.240         0   

     RankSeason  RankPlayoffs   OOBP   OSLG  
372         NaN           NaN  0.353  0.476  
339         NaN           NaN  0.350  0.480  
388         NaN           NaN  0.369  0.475  
399         NaN           NaN  0.384  0.499  
403         NaN           NaN  0.365  0.454  
249         NaN           NaN  0.372  0.471  
358         NaN           NaN  0.362  0.494  
417         NaN           NaN  0.371  0.448  
216         NaN           NaN  0.352  0.483  
182         NaN           NaN  0.357  0.462  
328         NaN           NaN  0.355  0.451  
209         NaN           NaN  0.349  0.447  
376         NaN           NaN  0.347  0.466  
309         NaN           NaN  0.349  0.466  
119         NaN           NaN  0.352  0.450  
176         NaN           NaN  0.358  0.467  
365         NaN           NaN  0.348  0.452  
218         NaN           NaN  0.362  0.457  
192         NaN           NaN  0.367  0.476  
92          NaN           NaN  0.353  0.476  
8           NaN           NaN  0.357  0.470  
298         NaN           NaN  0.360  0.473  
236         NaN           NaN  0.355  0.455  
285         NaN           NaN  0.348  0.462  
377         NaN           NaN  0.353  0.449  
400         NaN           NaN  0.349  

The teams we are left with are seemingly unexceptional, but the teams at the top have the potential to shine given their having been saved from many rounds of culling.

I would purchase HOU, team 372, and put my money into making sure that the players have the necessary utilities that they need to become exceptional.

In [171]:
display(df.iloc[0])

Team              HOU
League             NL
Year             2000
RS                938
RA                944
W                  72
OBP             0.361
SLG             0.477
BA              0.278
Playoffs            0
RankSeason        NaN
RankPlayoffs      NaN
OOBP            0.353
OSLG            0.476
Name: 372, dtype: object